lib/zlib.js

```diff
@@ -706,57 +706,72 @@ Zlib.prototype.params = function params(level, strategy, callback) {
 // generic zlib
 // minimal 2-byte header
 function Deflate(opts) {
-  if (!(this instanceof Deflate))
+  if (!(this instanceof Deflate)) {
```

Comment:
Note that this check will pass if you do `Deflate.call(someDeflateInstance)`.

The reliable way to check for `new` (/ `Reflect.construct`, which is equivalent) is check `if (!new.target)`.

Reply:
I opened a new pull-request for this change, thank you for the review @nicolo-ribaudo. It's always a pleasure reading your comments! https://github.com/nodejs/node/pull/54694